src/NDItk/MultigroupTable.hpp

```diff
@@ -48,12 +52,39 @@ class MultigroupTable {
   /**
    *  @brief Return the primary group structure record
    */
-  const multigroup::EnergyGroupStructure& structure() const { return this->structure_; }
+  const multigroup::EnergyGroupStructure& primaryGroupBoundaries() const { 
+    
+    return this->primary_structure_; 
+  }
+
+  /**
+   *  @brief Return the group structure record for an outgoing particle
+   */
+  const multigroup::EnergyGroupStructure& 
+  outgoingGroupBoundaries( unsigned int particle ) const {
+
+    auto pos = std::lower_bound( this->outgoing_structure_.begin(),
```

Comment:
How fast is something like this versus having the group boundaries stored in a std::vector?

Reply:
I could use a map to do this too. The lower_bound algorithm returns the position of the element that is either the one we are looking for or the first element that is larger than the one we are looking for (if that element is not there). It's basically binary search.

Since the number of particles is going to be small, I don't think it will matter performance wise.

Reply:
It should not. Between 1 and 5 particles.